xapian-core/tests/api_cluster.cc

```diff
@@ -80,3 +105,35 @@ DEFINE_TESTCASE(round_robin1, backend)
     }
     return true;
 }
+
+DEFINE_TESTCASE(stem_stopper1, backend)
+{
+    Xapian::Stem stemmer("english");
+    // By default, stemming strategy used is STEM_SOME
+    Xapian::StemStopper stopper(stemmer);
+    std::string term = "the";
+    stopper.add(term);
+    TEST(stopper(term));
+    TEST(stopper('Z' + stemmer(term)));
+
+    Xapian::StemStopper stopper_all_z(stemmer, Xapian::StemStopper::STEM_ALL_Z);
+    Xapian::StemStopper stopper_all(stemmer, Xapian::StemStopper::STEM_ALL);
+    term = "because";
+    stopper_all.add(term);
+    stopper_all_z.add(term);
+    TEST(!stopper_all_z(term));
+    TEST(stopper_all_z('Z' + stemmer(term)));
+    TEST(!stopper_all(term));
+    TEST(stopper_all('Z' + stemmer(term)));
```

Comment:
The `'Z' + ` here looks wrong to me - `stopper_all` should only stop the stemmed form of `"because"` **without** a `Z` prefix.  So as-is, this test should be failing I think (I can't tell without pulling the code and building it because there's a merge conflict and that seems to suppress CI runs - not really sure why).

As well as checking that the stoppers don't stop the unstemmed term and do stop the correctly prefixed stemmed term, it would be good to also check that each doesn't stop the incorrectly prefixed stemmed term - i.e. `stopper_all_z` doesn't stop when there's no `Z` prefix, and that `stopper_all` doesn't stop when there is.